front/components/assistant_builder/TeamSharingSection.tsx

```diff
@@ -1,280 +1,118 @@
 import {
-  BookOpenIcon,
-  Button,
-  CheckCircleIcon,
-  Dialog,
-  Icon,
+  ChevronDownIcon,
+  Chip,
+  DropdownMenu,
+  IconButton,
+  LockIcon,
+  PlanetIcon,
+  UserGroupIcon,
 } from "@dust-tt/sparkle";
 import type { AgentConfigurationScope, WorkspaceType } from "@dust-tt/types";
 import { isBuilder } from "@dust-tt/types";
-import { useRouter } from "next/router";
-import { useState } from "react";
 
+import { assistantUsageMessage } from "@app/lib/assistant";
+import { useAgentUsage } from "@app/lib/swr";
+
+/*
+ * Note: Non-builders cannot change to/from company assistant
+ */
 export function TeamSharingSection({
   owner,
   agentConfigurationId,
-  initialScope,
   newScope,
   setNewScope,
 }: {
   owner: WorkspaceType;
   agentConfigurationId: string | null;
-  initialScope: AgentConfigurationScope;
   newScope: Exclude<AgentConfigurationScope, "global">;
   setNewScope: (scope: Exclude<AgentConfigurationScope, "global">) => void;
 }) {
-  const [showPublishDialog, setShowPublishDialog] = useState(false);
-  const [showWorkspaceRemoveDialog, setShowWorkspaceRemoveDialog] =
-    useState(false);
-  const [showCreateDuplicateDialog, setShowCreateDuplicateDialog] =
-    useState(false);
-  const router = useRouter();
-
-  if (initialScope !== "private" && newScope === "private") {
-    throw new Error("Cannot change scope back to private");
-  }
-  if (initialScope === "global") {
-    return null;
-  }
+  const agentUsage = agentConfigurationId
```

Comment:
I don't think you can do this. You should never conditionally use a hook. Instead the recommended way is to make the swr url be null in cases where you don't want to fetch.

Reply:
See here: https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
Ack will change
Weird that I didn't get the Hooks error

Reply:
you would get it if the number / order of hooks change. So here, it's fairly unlikely that it gets re-rendered with a non-null `agentConfigurationId` after being rendered with a null `agentConfigurationId`

but it's still good practice to not do this

Reply:
Changed ! Good for you to merge?

Reply:
Yes all good 🙏